src/app/survey/upload/page.tsx

```diff
@@ -5,17 +5,77 @@ import { cn } from "~/lib/utils";
 
 import { UploadIcon } from "@radix-ui/react-icons";
 import { Loader2 } from "lucide-react";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { toast } from "~/components/ui/use-toast";
 import { newSurveyObject } from "~/app/survey/upload/newSurveyObject";
+import Papa from "papaparse";
+import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
 
 function SurveyUpload() {
+    const [surveyDate, setSurveyDate] = useState<Date>(new Date());
+    const [surveyName, setSurveyName] = useState<string>("");
     const { mutate, isPending, isSuccess, isError, error } =
         api.surveys.uploadNewSurvey.useMutation();
     const uploadFile = async (file: File) => {
         const fileContent = await file.text();
+        const parseResult = Papa.parse(fileContent, {
```

Comment:
Why do we move CSV parsing tot the client? This way its included in the JS bundle.